volume/local/local_unix.go

```diff
@@ -63,19 +68,26 @@ func setOpts(v *localVolume, opts map[string]string) error {
 	if len(opts) == 0 {
 		return nil
 	}
-	if err := validateOpts(opts); err != nil {
+	err := validateOpts(v, opts)
+	if err != nil {
 		return err
 	}
-
 	v.opts = &optsConfig{
 		MountType:   opts["type"],
 		MountOpts:   opts["o"],
 		MountDevice: opts["device"],
 	}
+	if val, ok := opts["size"]; ok {
+		size, err := units.RAMInBytes(val)
+		if err != nil {
+			return err
+		}
+		v.opts.Quota.Size = uint64(size)
+	}
 	return nil
 }
 
-func validateOpts(opts map[string]string) error {
+func validateOpts(v *localVolume, opts map[string]string) error {
```

Comment:
The idea here is to keep the function signature more generic, so that potential future options which need info from the volume don't have to add another one.
Passing in quotaEnabled would also just mean the check for `v.quotaCtl == nil` would move one function up, to then be passed in as bool. I think I prefer the current approach.

Though it does cause a bit of code-duplication between this and the actual quota-application in setOpts.
So maybe the check here can be removed entirely, and setOpts checks for quota support itself and bails if it's missing?

Reply:
Refactored this to get rid of the issue entirely and avoid the logic duplication.